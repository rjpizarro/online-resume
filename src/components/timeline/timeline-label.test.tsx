import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'

import TimelineLabel from "./timeline-label"

describe("TimelineLabel", () => {
  it("render normally with required props", () => {
    const tree = renderer
      .create(
        <TimelineLabel
          startDate={new Date("01/03/2014")}
          endDate={new Date("01/05/2015")}
          company="Test company"
          jobTitle="Test job title"
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("should format the dates properly", () => {
    const { getByTestId } = render(
      <TimelineLabel
        startDate={new Date("03/01/2014")}
        endDate={new Date("05/01/2015")}
        company="Test company"
        jobTitle="Test job title"
      />
    )

    expect(getByTestId("dates").innerHTML).toEqual("2014 Mar - 2015 May")
  })
})