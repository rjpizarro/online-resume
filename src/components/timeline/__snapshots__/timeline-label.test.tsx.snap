// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineLabel render normally with required props 1`] = `
<div>
  <span
    className="MuiTypography-root MuiTypography-caption"
    data-testid="dates"
  >
    2014 Jan
     - 
    2015 Jan
  </span>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1"
  >
    Test company
  </h6>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2"
  >
    Test job title
  </h6>
</div>
`;
